nav2_behavior_tree/plugins/decorator/is_path_longer_on_approach.cpp

```diff
@@ -0,0 +1,73 @@
+// Copyright (c) 2022 Neobotix GmbH
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <string>
+#include <memory>
+#include <vector>
+#include "nav2_util/geometry_utils.hpp"
+
+#include "nav2_behavior_tree/plugins/decorator/is_path_longer_on_approach.hpp"
+
+namespace nav2_behavior_tree
+{
+
+IsPathLongerOnApproach::IsPathLongerOnApproach(
+  const std::string & name,
+  const BT::NodeConfiguration & conf)
+: BT::DecoratorNode(name, conf)
+{
+  node_ = config().blackboard->get<rclcpp::Node::SharedPtr>("node");
+}
+
+inline BT::NodeStatus IsPathLongerOnApproach::tick()
+{
+  getInput("path", new_path_);
+  getInput("prox_leng", prox_leng_);
+  getInput("length_factor", length_factor_);
+
+  // Check if the old path is not same with the current one
+  if (new_path_ != old_path_ && old_path_.poses.size() != 0 && new_path_.poses.size() != 0) {
+    // Calculate and compare the old and the new path length, given the goal proximity
+    if ( (nav2_util::geometry_utils::calculate_path_length(new_path_, 0) >= prox_leng_) &&
```

Comment:
Wouldn't it be `nav2_util::geometry_utils::calculate_path_length(new_path_, 0) < prox_leng_` so that we only apply it when we're within the proximity to the goal?

Reply:
Yes, you are right. I will update it.